table/block_based_table_builder.cc

```diff
@@ -784,17 +784,27 @@ Status BlockBasedTableBuilder::Finish() {
                   &metaindex_block_handle);
 
     const bool is_data_block = true;
-    WriteBlock(index_blocks.index_block_contents, &index_block_handle,
-               !is_data_block);
+    if (!r->table_options.index_uncompressed) {
+      WriteBlock(index_blocks.index_block_contents, &index_block_handle,
+                 !is_data_block);
+    } else {
+      WriteRawBlock(index_blocks.index_block_contents, kNoCompression,
+                    &index_block_handle);
+    }
     // If there are more index partitions, finish them and write them out
     Status& s = index_builder_status;
     while (s.IsIncomplete()) {
       s = r->index_builder->Finish(&index_blocks, index_block_handle);
       if (!s.ok() && !s.IsIncomplete()) {
         return s;
       }
-      WriteBlock(index_blocks.index_block_contents, &index_block_handle,
-                 !is_data_block);
+      if (!r->table_options.index_uncompressed) {
+        WriteBlock(index_blocks.index_block_contents, &index_block_handle,
+                   !is_data_block);
```

Comment:
we should get rid of this variable and just pass false

Reply:
Agreed. Not sure why the code had this variable.